java/driver/flight-sql/src/main/java/org/apache/arrow/adbc/driver/flightsql/FlightSqlDriver.java

```diff
@@ -64,6 +64,11 @@ public AdbcDatabase open(Map<String, Object> parameters) throws AdbcException {
     } else {
       quirks = new SqlQuirks();
     }
-    return new FlightSqlDatabase(allocator, location, (SqlQuirks) quirks);
+    return new FlightSqlDatabase(
+        allocator,
+        location,
+        (SqlQuirks) quirks,
+        (String) parameters.get("adbc.username"),
+        (String) parameters.get("adbc.password"));
```

Comment:
It might be better to pass the `parameters` instead. The `FlightSqlDatabase` will need to know about all the params to decide how to build the `FlightClient`. For example, all the TLS options.

It would also be nice to have dataclasses or static objects with all the option values https://arrow.apache.org/adbc/main/driver/go/flight_sql.html#client-options